packages/web-components/src/components/va-text-input/va-text-input.tsx

```diff
@@ -54,7 +54,7 @@ export class VaTextInput {
   /**
    * The error message to render.
    */
-  @Prop() error?: string;
+  @Prop({ reflect: true }) error?: string;
```

Comment:
`reflect` will add the prop to the component's markup which is how the error styles are applied.